src/_base/harness/config/pipeline.yml

```
@@ -87,7 +87,7 @@ command('app deploy <environment>'):
     #!bash(harness:/helm)|=
     set -o pipefail
     cd "${ENVIRONMENT}"
-    doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml
+    (umask 0200 && doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml)
```

silly me, wrong way around in my last suggestion
```suggestion
    (umask 0077 && doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml)
```

---

Is this equivalent as the proposed solution from the issue: https://github.com/helm/helm/issues/9115#issuecomment-818832622 ?
> chmod 600 ~/.kube/config

---

yes, umask removes the bits that are allowed on a file on creation, so e.g. 0007 means if a file created with 666 permissions then it gets 660 permissions. You can see it easily with `(umask 0077 && touch testfile)` that it gets 600 permissions